src/main/webapp/script.js

```diff
@@ -55,9 +58,25 @@ let numMutations = 0;
 // (for ex, if a node was searched and the node was modified in the 1st, 4th, and 
 // 5th indices, this would be [1,4,5])
 let mutationIndexList = [];
+// An object representing a slider whose value can be changed by the user to 
+// modify currMutationIndex
+let mutationNumSlider;
+// The maximum number of mutations that can be applied to the initial graph
+// ie. the length of the mutations list
+let maxNumMutations = 0;
 
-// Stores total number of mutations, for display
-let totalNum = 0;
+/**
+ * Initializes the mutation index slider upon document load and sets it up to
+ * regenerate the graph when its value is changed
+ */
+function initializeSlider() {
+  mutationNumSlider = new MDCSlider(document.querySelector('.mdc-slider'));
+  mutationNumSlider.listen('MDCSlider:change', () => {
+    currMutationIndex = mutationNumSlider.value;
+    currMutationNum = (currMutationIndex === -1) ? -1 : mutationIndexList[currMutationIndex];
```

Comment:
do you need to make sure currMutationIndex < mutationIndexList.length (i mean I feel like that should that never happen but maybe as a precautionary measure haha)